node/src/tasks/bsp_upload_file.rs

```diff
@@ -450,6 +441,140 @@ where
             }
         };
 
+        let available_capacity = self
+            .storage_hub_handler
+            .blockchain
+            .query_available_storage_capacity(own_bsp_id)
+            .await
+            .map_err(|e| {
+                error!(
+                    target: LOG_TARGET,
+                    "Failed to query available storage capacity: {:?}", e
+                );
+                anyhow::anyhow!("Failed to query available storage capacity: {:?}", e)
+            })?;
+
+        // Increase storage capacity if the available capacity is less than the file size.
+        if available_capacity < event.size {
```

Comment:
This whole section of adding capacity is probably worth wrapping in a helper function, for better readability